internal/repository/employee/employee_mysql.go

```diff
@@ -0,0 +1,141 @@
+package employee
+
+import (
+	"ProyectoFinal/pkg/models"
+	"database/sql"
+	"fmt"
+)
+
+type mysqlRepository struct {
+	db *sql.DB
+}
+
+func NewMySQLRepository(db *sql.DB) Repository {
+	return &mysqlRepository{
+		db: db,
+	}
+}
+
+func (r *mysqlRepository) GetAll() ([]models.Employee, error) {
+	rows, err := r.db.Query(QueryGetAllEmployees)
+	if err != nil {
+		return nil, fmt.Errorf("error querying employees: %w", err)
+	}
+	defer rows.Close()
+
+	var employees []models.Employee
+	for rows.Next() {
+		var employee models.Employee
+		var warehouseID sql.NullInt64
+
+		err := rows.Scan(&employee.ID, &employee.CardNumberID, &employee.FirstName, &employee.LastName, &warehouseID)
+		if err != nil {
+			return nil, fmt.Errorf("error scanning employee row: %w", err)
+		}
+
+		if warehouseID.Valid {
+			employee.WarehouseID = int(warehouseID.Int64)
+		}
+
+		employees = append(employees, employee)
+	}
+
+	if err = rows.Err(); err != nil {
+		return nil, fmt.Errorf("error iterating employee rows: %w", err)
+	}
+
+	return employees, nil
+}
+
+func (r *mysqlRepository) GetById(id int) (models.Employee, error) {
+	row := r.db.QueryRow(QueryGetEmployeeById, id)
+
+	var employee models.Employee
+	var warehouseID sql.NullInt64
+
+	err := row.Scan(&employee.ID, &employee.CardNumberID, &employee.FirstName, &employee.LastName, &warehouseID)
+	if err != nil {
+		if err == sql.ErrNoRows {
```

Comment:
Creo que esa validación no estaria funcionando, lo ideal seria usar errors.is
```suggestion
				if errors.Is(err, sql.ErrNoRows) {
			newError := pkgErrors.WrapErrNotFound("Employee", "id", id)
			return nil, newError
		}
```